src/cargo/core/workspace.rs

```diff
@@ -679,7 +679,9 @@ impl<'cfg> Workspace<'cfg> {
             for path in default {
                 let normalized_path = paths::normalize_path(&path);
                 let manifest_path = normalized_path.join("Cargo.toml");
-                if !self.members.contains(&manifest_path) {
+                if !self.members.contains(&manifest_path)
+                    && (self.is_virtual() || manifest_path != root_manifest_path)
```

Comment:
I feel like this is prone to us having similar errors elsewhere.  Should we instead add the root manifest to `self.members` when in a non-virtual workspace?

Of course , there is the chance that this could possibly break something else.  I'm somewhat hopeful it won't because non-workspace packages have themselves as a member (and as a default).